### ISSUE TITLE
[ANALYTICS] M&A deal screening: add logistic regression + cross-validated classifier

### ISSUE THREAD

**Issue by DogInfantry** [DogInfantry/capital-markets-intelligence#6]
## Summary
Replace the current correlation-only approach in the M&A deal screening model with a proper logistic regression (and optionally a gradient-boosted classifier) with cross-validation.

## Motivation
Point-biserial correlations show which features are *associated* with completion, but can't produce calibrated completion *probabilities* — which is what deal teams actually use. A logistic model is the standard tool in practitioners' deal screening frameworks (referenced in PwC Deals, Deutsche Bank M&A advisory).

### Model 1: Logistic Regression (primary)
```python
from sklearn.linear_model import LogisticRegression
from sklearn.model_selection import StratifiedKFold, cross_val_score
```
- Features: deal size log, cross_border flag, premium_pct, sector dummies, deal type dummies
- Target: `completed` (binary)
- 5-fold stratified cross-validation
- Report: AUC-ROC, accuracy, confusion matrix

### Model 2: Gradient-Boosted Trees (optional, bonus)
- `sklearn.ensemble.GradientBoostingClassifier`
- SHAP feature importance chart

### Keep existing:
- Point-biserial correlation table (keep as an exploratory view)
- Feature importance waterfall chart in Plotly dashboard

## Output Changes
- `data/processed/deal_screening_model.csv` gains: `completion_prob_lr`, `completion_prob_gbt`
- New Excel tab: "Classification Model" with AUC, CV scores, confusion matrix
- Dashboard gains: ROC curve panel, probability scatter

## Acceptance Criteria
- [ ] Logistic regression trains without errors on the 20-deal dataset
- [ ] AUC-ROC reported (note: small sample — interpret with caution, add a disclaimer)
- [ ] Feature importance table updated
- [ ] Methodology note added to `docs/`

## Notes
- 20 deals is a very small sample — the model is illustrative, not predictive. Add a clear disclaimer in outputs.
- `sklearn` not currently in requirements.txt — add it.
